ggml/src/lib.rs

```diff
@@ -466,3 +466,49 @@ fn i32_to_usize(val: i32) -> usize {
 fn i64_to_usize(val: i64) -> usize {
     usize::try_from(val).unwrap()
 }
+
+/// Quantizes `src` into `dst` using `q4_0` quantization.
+///
+/// # Safety
+///
+/// You must ensure the arrays passed in are of the correct size.
```

Comment:
We should know the correct sizes here, right? I'd really suggest just adding a couple asserts to make sure it's correct, or ideally have the functions return `Result` so it can be handled more gracefully than just panicking.

Either way is better than possibly allowing memory errors though. Same thing for `quantize_q4_1`.

Reply:
Yes, that's something I added as I wasn't sure when I was looking over the code: we should definitely encode these constraints more directly now that we have a CLI tool that uses them.

Reply:
Another thing I did in my fork is change the view functions to take the number of elements rather than bytes. It doesn't seem like there's an actual use to having to pull up and multiply by the element size all the time. You'd never want to offset into half of a float or something, right?

Reply:
Further discussion is in the PR #128